// temboo/Library/Labs/Social.js

var util = require("util");
var choreography = require("temboo/core/choreography");


/*
    GetContacts
    Retrieves your social contacts from multiple APIs in one API call.
*/


var GetContacts = function(session) {
    /*
        Create a new instance of the GetContacts Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Labs/Social/GetContacts"
    GetContacts.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetContactsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetContactsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetContacts
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetContactsInputSet = function() {
    GetContactsInputSet.super_.call(this);
        /*
        Set the value of the APICredentials input for this Choreo. ((conditional, json) A list of credentials for the APIs you wish to access. See Choreo documentation for formatting examples.)
        */
        this.set_APICredentials = function(value) {
            this.setInput("APICredentials", value);
        }

        /*
        Set the value of the ScreenName input for this Choreo. ((conditional, string) The Twitter screen name to retrieve followers for.)
        */
        this.set_ScreenName = function(value) {
            this.setInput("ScreenName", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetContacts Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetContactsResultSet = function(resultStream) {
    GetContactsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. ((json) Contains the merged results from the API responses.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
}

util.inherits(GetContacts, choreography.Choreography);
util.inherits(GetContactsInputSet, choreography.InputSet);
util.inherits(GetContactsResultSet, choreography.ResultSet);
exports.GetContacts = GetContacts;


/*
    UpdateAllStatuses
    Shares a post across multiple social networks such as Facebook, Tumblr, and Twitter.
*/


var UpdateAllStatuses = function(session) {
    /*
        Create a new instance of the UpdateAllStatuses Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Labs/Social/UpdateAllStatuses"
    UpdateAllStatuses.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new UpdateAllStatusesResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new UpdateAllStatusesInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the UpdateAllStatuses
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var UpdateAllStatusesInputSet = function() {
    UpdateAllStatusesInputSet.super_.call(this);
        /*
        Set the value of the APICredentials input for this Choreo. ((required, json) A list of credentials for the APIs you wish to access. See Choreo documentation for formatting examples.)
        */
        this.set_APICredentials = function(value) {
            this.setInput("APICredentials", value);
        }

        /*
        Set the value of the Message input for this Choreo. ((required, string) The message to be posted across social networks.)
        */
        this.set_Message = function(value) {
            this.setInput("Message", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the UpdateAllStatuses Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var UpdateAllStatusesResultSet = function(resultStream) {
    UpdateAllStatusesResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. ((json) A list of results for each API.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
}

util.inherits(UpdateAllStatuses, choreography.Choreography);
util.inherits(UpdateAllStatusesInputSet, choreography.InputSet);
util.inherits(UpdateAllStatusesResultSet, choreography.ResultSet);
exports.UpdateAllStatuses = UpdateAllStatuses;
